File: HomeWork1/Hw1/Calculator/Calculator.cs
namespace Hw1.Calculator
{
public static class Calculator
{
    public static double Calculate(int val1, CalculatorOperation operation, int val2)
    {
        var result = operation switch
        {
            CalculatorOperation.Divide => (double) val1 / val2,
            CalculatorOperation.Multiply => val1 * val2,
            CalculatorOperation.Minus => val1 - val2,
            _ => val1 + val2,
        };
            
        return result;
    }
}
}

File: HomeWork6/Homework6/Calculator.fs
module Giraffe.Calculator

let Calculate (val1: decimal, operation, val2: decimal) =
    match operation with
    | Divide ->
        match val2 with
        | 0m -> Error ErrorType.DivideByZero
        | _ ->  val1 / val2 |> Ok 
    | Plus -> val1 + val2 |> Ok
    | Minus -> val1 - val2 |> Ok
    | Multiply -> val1 * val2 |> Ok